python3Packages.pyipv8: 3.0.2197-unstable-2025-07-29 -> 3.1; tribler: 8.2.3 -> 8.2.3-unstable-2025-10-14

mdaniels5757 opened the pull request:
Diffs:
* https://github.com/Tribler/tribler/compare/v8.2.3...3e5bc56a15c568d0ba41262cad63155445e062da
* https://github.com/Tribler/py-ipv8/compare/db39b85f4c28880dee24d1b59d8eae8ca8b9c03d...3.1

ipv8 release notes: https://github.com/Tribler/py-ipv8/releases/tag/3.1

Closes: #439047

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [X] x86_64-darwin
  - [X] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [X] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [X] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mdaniels5757 committed:
python3Packages.pyipv8: 3.0.2197-unstable-2025-07-29 -> 3.1

Diff: https://github.com/Tribler/py-ipv8/compare/db39b85f4c28880dee24d1b59d8eae8ca8b9c03d...3.1

Release notes: https://github.com/Tribler/py-ipv8/releases/tag/3.1

mdaniels5757 commented:
It was a draft because tribler's released version requires a version of pyipv8 that is incompatible with this. I just tried updating tribler and pyipv8 to HEAD, but tribler will need #448414 for that to work, I think.

SuperSandro2000 commented on pkgs/development/python-modules/pyipv8/default.nix:
```diff
@@ -15,16 +15,16 @@
   pname = "pyipv8";
-  version = "3.0.2197-unstable-2025-07-29";
+  version = "3.1";
   format = "setuptools";
```
Can we convert to pyproject = true?

mdaniels5757 replied:
Sure, done

mdaniels5757 committed:
tribler: 8.2.3 -> 8.2.3-unstable-2025-10-14

Diff: https://github.com/Tribler/tribler/compare/v8.2.3...3e5bc56a15c568d0ba41262cad63155445e062da
